moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/cost_convergence_termination_condition.h

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <ompl/base/PlannerTerminationCondition.h>
+namespace ompl_interface
+{
+namespace ob = ompl::base;
+class CostConvergenceTerminationCondition : public ob::PlannerTerminationCondition
+{
+public:
+  CostConvergenceTerminationCondition(size_t solutions_window = 10, double convergence_threshold = 0.9)
+    : ob::PlannerTerminationCondition([] { return false; })
+    , solutions_window_(solutions_window)
+    , convergence_threshold_(convergence_threshold)
+  {
+    std::cout << " test" << std::endl;
+  }
+
+  // Compute the running average cost of the last solutions (solution_window) and terminate if the cost of a
+  // new solution is higher than convergence_threshold times the average cost.
+  void processNewSolution(const std::vector<const ob::State*>& solution_states, const ob::Cost solution_cost)
+  {
+    ++solutions_;
+    size_t solutions = std::min(solutions_, solutions_window_);
+    double new_cost = ((solutions - 1) * average_cost_ + solution_cost.value()) / solutions;
+    double cost_ratio = average_cost_ / new_cost;
+    average_cost_ = new_cost;
+    if (solutions == solutions_window_ && cost_ratio > convergence_threshold_)
```

Comment:
Shouldn't this be `cost_ratio < convergence_threshold_`? If the cost ratio is still very high (i.e. the new cost is descending sharply), doesn't that imply that more improvements are likely to be found, and when the cost ratio is lower, that you're approaching a optimum?

Reply:
Yes of course! I realize this I pushed an outdated version, I'll get to it first thing next week.

Reply:
this is still slightly confusing (but should work) as I would expect the threshold to be some small number (`epsilon`) and then check `1.0 - cost_ratio < epsilon_'

also I'm not sure if the comment and the code describe the same thing?

Reply:
@simonschmeisser actually, the comment is a bit misleading but the check is actually the same:
`new_cost / average_cost > convergence_threshold`
equals
`new_cost > convergence_threshold * average_cost` since average_cost is positive. I'll change that anyway. Your version with epsilon somewhat does the same thing, I just thought that a threshold that matches the ratio is easier to comprehend.